Feature Request - Bring back ConsoleShell functionallity

Hey guys,
In my product team we created a custom powershell-based shell in as a way to controll our cloud service.
We did this based on the ConsoleShell object and customised RunspaceConfiguration.

As a part of moving to .Net Core we started examing the posibillity of moving our shell to powershell core but as we all know RunspaceConfiguration does not exist in the core version and there is no way to pass InitialSessionState to the ConsoleShell.Start() method.

if we could pass a initialSessionState to the Start method that would be great!

Thanks,
Asaf.

There might be a problem with the nupkg for ConsoleHost.dll.  I tried a simple app and had to directly ref the assembly to get it to work, but still got this warning:

```none
PS> dotnet publish --runtime osx-x64
Microsoft (R) Build Engine version 16.4.0-preview-19502-03+3af680463 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 1.06 sec for /Users/steve/test/consoleshell/consoleshell.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
CSC : warning CS8012: Referenced assembly 'Microsoft.PowerShell.ConsoleHost, Version=6.2.3.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' targets a different processor. [/Users/steve/test/consoleshell/consoleshell.csproj]
  consoleshell -> /Users/steve/test/consoleshell/bin/Debug/netcoreapp2.1/osx-x64/consoleshell.dll
  consoleshell -> /Users/steve/test/consoleshell/bin/Debug/netcoreapp2.1/osx-x64/publish/
```

For now, to get unblocked, try referencing the assembly directly:

```xml
    <Reference Include="Microsoft.PowerShell.ConsoleHost">
      <HintPath>/Users/steve/.nuget/packages/microsoft.powershell.consolehost/6.2.3/runtimes/osx/lib/netcoreapp2.1/Microsoft.PowerShell.ConsoleHost.dll</HintPath>
    </Reference>
```

cc @adityapatwardhan

--- a/test/hosting/test_HostingBasic.cs
+++ b/test/hosting/test_HostingBasic.cs
@@ -179,7 +179,7 @@ namespace PowerShell.Hosting.SDK.Tests
         [Fact]
         public static void TestConsoleShellScenario()
         {
-            int ret = ConsoleShell.Start("Hello", "", new string[] { "-noprofile", "-c", "exit 42" });
+            int ret = ConsoleShell.Start("Hello", string.Empty, new string[] { "-noprofile", "-c", "exit 42" });
             Assert.Equal(42, ret);
         }
     }

--- a/test/hosting/test_HostingBasic.cs
+++ b/test/hosting/test_HostingBasic.cs
@@ -6,6 +6,7 @@ using System.IO;
 using System.Management.Automation;
 using System.Security;
 using Microsoft.Management.Infrastructure;
+using Microsoft.PowerShell;
 using Xunit;
 
 namespace PowerShell.Hosting.SDK.Tests
@@ -173,6 +174,13 @@ namespace PowerShell.Hosting.SDK.Tests
                 Assert.Equal("Joe", foo.Name);
                 Assert.Equal("Unknown", foo.Path);
             }
+        }
+
+        [Fact]
+        public static void TestConsoleShellScenario()
+        {
+            int ret = ConsoleShell.Start("Hello", "", new string[] { "-noprofile", "-c", "exit 42" });
+            Assert.Equal(42, ret);
         }
     }
 }

--- a/tools/packaging/packaging.psm1
+++ b/tools/packaging/packaging.psm1
@@ -1745,11 +1745,15 @@ function CopyReferenceAssemblies
         [string[]] $assemblyFileList
     )
 
+    $supportedRefList = @(
+        "Microsoft.PowerShell.Commands.Utility",
+        "Microsoft.PowerShell.ConsoleHost")
+
     switch ($assemblyName) {
-        "Microsoft.PowerShell.Commands.Utility" {
-            $ref_Utility = Join-Path -Path $refBinPath -ChildPath Microsoft.PowerShell.Commands.Utility.dll
-            Copy-Item $ref_Utility -Destination $refNugetPath -Force
-            Write-Log "Copied file $ref_Utility to $refNugetPath"
+        { $_ -in $supportedRefList } {
+            $refDll = Join-Path -Path $refBinPath -ChildPath "$assemblyName.dll"
+            Copy-Item $refDll -Destination $refNugetPath -Force
+            Write-Log "Copied file $refDll to $refNugetPath"
         }
 
         "Microsoft.PowerShell.SDK" {
@@ -1931,7 +1935,8 @@ function New-ReferenceAssembly
     $SMAReferenceAssembly = $null
     $assemblyNames = @(
         "System.Management.Automation",
-        "Microsoft.PowerShell.Commands.Utility"
+        "Microsoft.PowerShell.Commands.Utility",
+        "Microsoft.PowerShell.ConsoleHost"
     )
 
     foreach ($assemblyName in $assemblyNames) {
@@ -2119,10 +2124,8 @@ function GenerateBuildArguments
     $arguments += "/p:RefAsmVersion=$RefAssemblyVersion"
     $arguments += "/p:SnkFile=$SnkFilePath"
 
-    switch ($AssemblyName) {
-        "Microsoft.PowerShell.Commands.Utility" {
-            $arguments += "/p:SmaRefFile=$SMAReferencePath"
-        }
+    if ($AssemblyName -ne "System.Management.Automation") {
+        $arguments += "/p:SmaRefFile=$SMAReferencePath"
     }
 
     return $arguments

--- a/tools/packaging/projects/reference/Microsoft.PowerShell.ConsoleHost/Microsoft.PowerShell.ConsoleHost.csproj
+++ b/tools/packaging/projects/reference/Microsoft.PowerShell.ConsoleHost/Microsoft.PowerShell.ConsoleHost.csproj
@@ -1,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <Version>$(RefAsmVersion)</Version>
+    <DelaySign>true</DelaySign>
+    <AssemblyOriginatorKeyFile>$(SnkFile)</AssemblyOriginatorKeyFile>
+    <SignAssembly>true</SignAssembly>
+    <LangVersion>Latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="System.Management.Automation">
+    <HintPath>$(SmaRefFile)</HintPath>
+    </Reference>
+  </ItemGroup>
+</Project>

